### PR TITLE
Check if a named object exists *before* trying to create it.

### PR DESCRIPTION
**Pull request by thetorpedodog**
TileDB Cloud used to forbid objects with duplicate names (e.g., you couldn't have two arrays with the same name). However, because distinct groups may contain entries with the same name, this is no longer the case. For this reason, we need to check for an object's existence at a given name before attempting to create a new one.

---

Tested locally with success. Fixes [sc-29723].

### CODE REVIEW

**Review by ihnorton — APPROVED** [review]
Thanks. Not tested locally, but LGTM as long as this fixes the registration of the notebook-array as an _array_ rather than a notebook (I suspect that was due to applying metadata updates on the wrong array?)

**Comment by thetorpedodog** [comment]
> Thanks. Not tested locally, but LGTM as long as this fixes the registration of the notebook-array as an _array_ rather than a notebook (I suspect that was due to applying metadata updates on the wrong array?)

tl;dr it would create a new array with the same `tiledb://me/notebook-name` name as the existing array, so the "this is a notebook" metadata would *never* get written to the new array since it couldn't ever be addressed by that name.
